### ISSUE TITLE
Android SDK - Airship Contact Management | Contact subscriptions is carried forward from one user contact to another user contact

### ISSUE THREAD

**Issue by Angad-Tiwari** [urbanairship/react-native-airship#494]
# Preliminary Info

### What Airship dependencies are you using?
`"@ua/react-native-airship": "^15.2.1"`

### What are the versions of any relevant development tools you are using?
```
IDE - Visual Studio Code 
Version: 1.76.2 (Universal)
Commit: ee2b180d582a7f601fa6ecfdad8d9fd269ab1884
Date: 2023-03-14T17:54:09.061Z
Electron: 19.1.11
Chromium: 102.0.5005.196
Node.js: 16.14.2
V8: 10.2.154.26-electron.0
OS: Darwin x64 22.3.0
Sandboxed: No
```
```
"react": "^17.0.2",
"react-native": "^0.68.2"
```

# Report

### What are the steps to reproduce the unexpected behavior?
Issue replicated on Android Platform only, iOS app is working fine

                                                      Subscription List
                                                  #1 = my_insights
                                                  #2 = weekend_roundup
                                                  #3 = mckinsey_explainers
                                                  #4 = author_talks
                                                  #5 = the_mckinsey_podcast
                                                  #6 = building_an_inclusive
                                                  #7 = diversity_inclusion

                                                       Steps Followed
                                        1. Signin > ben.stone91@mailinator.com (named-user-id = 14222771)
                                        2. Default Contact Subscription > ALL
                                        3. User Unsubscribed > #1, #3, #5, #7
                                             - So, the remaining is Subscribed > #2, #4, #6 only
                                        4. Signout
                                        5. Register > ben.stone92@mailinator.com (named-user-id = 14222772)
                                        6. Default Contact Subscription > #1, #2, #4, #6

                                        Actual Output   - #1, #2, #4, #6
                                        Expected Output - #1 (we are subscribing to #1 = my_insights by default for the new registered user)

- Checked the Subscription List under 'Airship Console > Contact Management' and shows the correct Subscription List i.e. my_insights (only)
     - https://go.airship.com/apps/WU7qChP5TTyBpxxKWK-Cjw/contact_management/?q=14222772

I do see some caching mechanism under com.urbanairship.contacts.Contact.java, with 10 min. caching age.
```private static final long SUBSCRIPTION_CACHE_LIFETIME_MS = 10 * 60 * 1000; // 10M
    private static final long SUBSCRIPTION_LOCAL_HISTORY_CACHE_LIFETIME_MS = 10 * 60 * 1000; // 10M
    private final CachedValue<Map<String, Set<Scope>>> subscriptionListCache;
    private final List<CachedValue<ScopedSubscriptionListMutation>> subscriptionListLocalHistory;
```

Is something buggy here? Because after a few minutes, when I again open the 'Notification Pref Screen', it shows Expected Output.

### What unexpected behavior are you seeing?
For the newly registered user (let's say User-B) expected only 'my_insights' as subscribed, but instead, last contact subscription (User-A) is been carried forward

### What is the expected behavior?
For the newly registered users (lets say User-B) expected only 'my_insights' as subscribed

### Do you have logging for the issue?
[log.log](https://github.com/urbanairship/react-native-airship/files/11118337/log.log)

### Other info
Within the Android app,  every time user sign in, register or logout, the named user has been set and reset using the Urban Airship API.
UAirship.shared().contact.identify(userId)
UAirship.shared().contact.reset()

Expecting API will clear the contact subscriptions cache on `UAirship.shared().contact.reset()`

**Comment by Angad-Tiwari** [comment]
@ralepinski did you get a chance to look into it?

**Comment by rlepinski** [comment]
@Angad-Tiwari could you link me to the preference center definition? Are these channel subscription items or contact subscription items that are showing the wrong value?

**Comment by Angad-Tiwari** [comment]
@rlepinski 
https://docs.airship.com/platform/mobile/preference-center/getting-started/?groupid=android-kotlin

Contact subscription items are the ones that show the wrong values.
FYI, named-user is enabled from Airship console.
using the latest "@ua/react-native-airship": "^15.2.1" react node module

**Comment by rlepinski** [comment]
I think I see the issue with your additional info through support, its a different then what I thought it would. I believe its from the local history cache we store on the device. On iOS we store it with the contact ID and only apply the right one. On Android we do not do that. I believe it should be a simple fix but will require an Android SDK release first before I can address it in this plugin.

**Comment by rlepinski** [comment]
Fixed in 15.2.2
